sky/utils/context_utils.py

```diff
@@ -0,0 +1,168 @@
+"""Utilities for SkyPilot context."""
+import asyncio
+import functools
+import io
+import multiprocessing
+import os
+import subprocess
+import sys
+from typing import Any, Callable, IO, Optional, Tuple
+
+from sky import sky_logging
+from sky.utils import context
+from sky.utils import subprocess_utils
+
+StreamHandler = Callable[[IO[Any], IO[Any]], str]
+
+
+# TODO(aylei): call hijack_sys_attrs() proactivly in module init once we have
+# context widely adopted.
+def hijack_sys_attrs():
```

Comment:
This will only be used on the server-side, right? We should definitely avoid doing this when skypilot is being used as a python library.

Reply:
Good call, currently it is only called in the uvicorn handler process. I think the comment should be updated to reflect the library concern.